### PR TITLE
Issue #21642: Remove in-progress media tab from homescreen

### PR DESCRIPTION
**Pull request by jonalmeida**
Not removing the other logic because we're going to re-add this after some refinement in the future.

### To download an APK when reviewing a PR:
1. click on Show All Checks,
2. click Details next to "Taskcluster (pull_request)" after it appears and then finishes with a green checkmark,
3. click on the "Fenix - assemble" task, then click "Run Artifacts".
4. the APK links should be on the left side of the screen, named for each CPU architecture

### CODE REVIEW

**Review by gabrielluong — APPROVED** [review]

File: app/src/main/java/org/mozilla/fenix/ext/BrowserState.kt [hunk 1]
```diff
@@ -24,16 +24,9 @@ import kotlin.math.max
 fun BrowserState.asRecentTabs(): List<RecentTab> {
     return mutableListOf<RecentTab>().apply {
         val lastOpenedNormalTab = lastOpenedNormalTab
-        val inProgressMediaTab = inProgressMediaTab
```
Comment (gabrielluong) [inline app/src/main/java/org/mozilla/fenix/ext/BrowserState.kt:27]: We could also wrap this behind a feature flag?

Comment (jonalmeida) [reply app/src/main/java/org/mozilla/fenix/ext/BrowserState.kt:27]: Yeah, I wasn't sure to what extent we should be "removing it". So I opted for this solution so we can land it before the beta cut.

Comment (jonalmeida) [reply app/src/main/java/org/mozilla/fenix/ext/BrowserState.kt:27]: Another reason I forgot about, is that we get intermittent test failures when we have feature flags here. See https://github.com/mozilla-mobile/fenix/pull/21557.

Comment (csadilek) [reply app/src/main/java/org/mozilla/fenix/ext/BrowserState.kt:27]: Let's land as is. Thought about this last week and it's a trade-off. We don't want feature flags to be too fine-grained, so arguably this is a change in behaviour of an existing features as opposed to a feature itself. We can revert this to enable again when we want it (have time to add media controls).

**Review by csadilek — APPROVED** [review]

**Pull request merged**